### PR TITLE
internal/telemetry: fix flaky client test

### PR DESCRIPTION
**Pull request by nsrip-dd**
TestClient was a little over-complicated. It expected the client
messages to come in a specific order. However, it is entirely possible
for the start message, which is sent in a background goroutine, to get
scheduled/show up *after* the heartbeat and stop messages. Make the test
simpler and not scheduler-dependent by just checking that the client
starts up and we get a heartbeat.

### CODE REVIEW

File: internal/telemetry/client_test.go [hunk 1]
```diff
@@ -13,60 +13,38 @@ import (
...
 )
 
 func TestClient(t *testing.T) {
-	ch := make(chan telemetry.RequestType)
-	var (
-		hb           sync.WaitGroup
-		gotheartbeat int64
-	)
-	hb.Add(1) // signal that we got a heartbeat
+	heartbeat := make(chan struct{})
```
Comment (felixge) [inline internal/telemetry/client_test.go:23]: NIT: I usually use this to reduce the chances of leaking a goroutine that's trying to send on the channel if the reader goroutine exists unexpectedly. This does not seem to be an issue with this particular test, so feel free to ignore.

```suggestion
	heartbeat := make(chan struct{}, 1)
```

**Review by felixge — APPROVED** [review]
LGTM!

File: internal/telemetry/client_test.go (hunk 1 above)
Comment (nsrip-dd) [reply internal/telemetry/client_test.go:23]: Yes, I opted for a non-blocking send in the server handler for the same reason of avoiding a leak. 👍

Comment (felixge) [reply internal/telemetry/client_test.go:23]: Yeah, I saw that – thanks for thinking about these edge cases! The only advantage of (ab)using the channel capacity for this is that it's less code 🙈

**Review by dianashevchenko — APPROVED** [review]

**Pull request merged**